core/src/parking_lot.rs

```diff
@@ -28,6 +31,15 @@ use smallvec::SmallVec;
 // around the stdio file descriptors.
 // Here we use #[linkage] and #[export_name] to make both versions point to
 // the same instance.
+//
+// REVIEW: Hm yeah something else will be needed here, we shouldn't be dealing
+// with mangling like this and/or symbol linkage, it's just asking for problems.
+// It's long been thought that unit tests of libstd itself are just way too
+// difficult to deal with and this could be the nail in the coffin. If this
+// can't be fixed by other means then all unit tests in libstd should be
+// extracted to standalone tests in `src/libstd/tests` in the same manner as
+// `src/libcore/tests` and then unit tests for the libstd library should be
+// disabled.
```

Comment:
When doing this, how do you deal with privacy issues?

Reply:
NB. Using a mangled name is not necessary, it just needs to be a name.

Reply:
You wouldn't, or it would be public and unstable. It's not great either way but this isn't gonna fly as-is.

Reply:
Maybe something like [this](https://github.com/rust-lang/rust/blob/589beb979c0f1a9736e02e578913db88398c3ba5/src/libstd/panicking.rs#L231) could also be used here?